libraries/resource_provider.rb

```diff
@@ -183,6 +183,45 @@ def create(method = :create)
       end
     end
 
+    # Adds scopes to the Oneview resource if scope is not already added
+    def add_to_scopes
+      apply_scopes_action(:add_to_scopes, :add_scope) { |scope| @item['scopeUris'].include?(scope['uri']) }
+    end
+
+    # Removes scopes from the Oneview resource if scope is already added
+    def remove_from_scopes
+      apply_scopes_action(:remove_from_scopes, :remove_scope) { |scope| !@item['scopeUris'].include?(scope['uri']) }
+    end
+
+    # Helper method to apply method related to add or remove scopes in Oneview resource
+    def apply_scopes_action(action, resource_method, &ignore_scope_if)
+      return if @new_resource.scopes.nil? || @new_resource.scopes.empty?
```

Comment:
return a log informing it was skipped because no scope was specified:

Something like:
```ruby
return Chef::Log.info('No scopes were specified. Skipping this update') if ...
```